src/main/java/com/rultor/agents/ec2/AmazonEC2.java

```
@@ -0,0 +1,146 @@
+    /**
+     * EC2 instance type (like <i>t1.micro</i>, <i>m1.small</i> etc.).
+     */
+    private final transient String type;
+    /**
+     * EC2 instance key pair name.
+     */
+    private final transient String keyname;
+
+    /**
+     * Public ctor.
+     * @param cred Credentials for AWS
+     * @param zone AWS region
+     * @param instype Instance type
+     * @param key Instance key pair name
+     * @checkstyle ParameterNumberCheck (12 lines)
+     */
+    public AmazonEC2(
+        @NotNull final AWSCredentials cred,
+        @NotNull final String zone,
+        @NotNull final String instype,
+        @NotNull final String key) {
+        super();
+        this.credentials = cred;
+        this.region = zone;
+        this.type = instype;
+        this.keyname = key;
+    }
+
+    @Override
+    public final Instance runOnDemand() throws IOException {
+        Instance instance;
+        final AmazonEC2Client client = new AmazonEC2Client(
+            this.credentials
+        );
+        client.setEndpoint(this.region);
+        final List<Instance> instances = client.runInstances(
+            this.create()
+        ).getReservation().getInstances();
+        if (instances.isEmpty()) {
+            throw new IllegalStateException(
+                "No Instance was available in the RunInstanceRequest reply!"
+            );
+        } else {
+            instance = instances.get(0);
+        }
+        InstanceState state = instance.getState();
+        if (state.getCode() < RUNNING_STATE) {
+            final DescribeInstanceStatusRequest statusreq =
+                new DescribeInstanceStatusRequest();
+            statusreq.withInstanceIds(instance.getInstanceId());
+            while (state.getCode() < RUNNING_STATE) {
+                try {
+                    TimeUnit.SECONDS.sleep(Tv.FIFTEEN);
+                } catch (final InterruptedException ex) {
+                    Thread.currentThread().interrupt();
+                    break;
```

why swallowing the exception? why can't we just throw it up, wrapped into `IllegalStateException`